src/main/java/net/minecraftforge/common/extensions/IForgePlayer.java

```diff
@@ -22,74 +23,118 @@ private Player self()
-     * Checks if the player can reach (right-click) the passed entity.<br>
-     * On the server, additional leniency is added to account for movement/lag.
-     * @param entity The entity being range-checked.
+     * Checks if the player can reach a block.<br>
+     * On the server, additional padding is added to account for movement/lag.
+     * @param pos The position being range-checked.
      * @param padding Extra validation distance.
-     * @return If the player can interact with the passed entity.
+     * @return If the player can interact with this location.
      */
-    default boolean canInteractWith(Entity entity, double padding)
+    default boolean canReach(BlockPos pos, double padding)
     {
-        return isCloseEnough(entity, getReachDistance() + padding);
+        double reach = this.getBlockReach() + padding;
+        return self().getEyePosition().distanceToSqr(pos.getX() + 0.5D, pos.getY() + 0.5D, pos.getZ() + 0.5D) < reach * reach;
     }
 
     /**
-     * Utility check to see if the player is close enough to a target entity.
+     * Utility check to see if the player is close enough to a target entity. Uses "eye-to-closest-corner" checks.
+     * Vanilla used "bottom-center-to-center" checks, so these are more accurate, and work better with large AABB's.
      * @param entity The entity being checked against
      * @param dist The max distance allowed
      * @return If the eye-to-center distance between this player and the passed entity is less than dist.
      */
     default boolean isCloseEnough(Entity entity, double dist)
     {
         Vec3 eye = self().getEyePosition();
-        Vec3 targetCenter = entity.getPosition(1.0F).add(0, entity.getBbHeight() / 2, 0);
-        Optional<Vec3> hit = entity.getBoundingBox().clip(eye, targetCenter); //This hit should almost always be present, but we have a fallback just in case.  It likely indicates lag if it is not present.
-        return (hit.isPresent() ? eye.distanceToSqr(hit.get()) : self().distanceToSqr(entity)) < dist * dist;
+        AABB aabb = entity.getBoundingBox().inflate(entity.getPickRadius());
+        Vec3 target = new Vec3(Mth.clamp(eye.x, aabb.minX, aabb.maxX), Mth.clamp(eye.y, aabb.minY, aabb.maxY), Mth.clamp(eye.z, aabb.minZ, aabb.maxZ));
+        return eye.distanceToSqr(target) < dist * dist;
     }
 
+    // Deprecated / for binary compat - to be removed in next BC cycle
+
     /**
-     * Checks if the player can reach (right-click) a block.<br>
-     * On the server, additional leniency is added to account for movement/lag.
-     * @param pos The position being range-checked.
-     * @param padding Extra validation distance.
-     * @return If the player can interact with this location.
+     * @deprecated Use {@link #getEntityReach()}
+     */
+    default double getAttackRange()
+    {
+        return this.getEntityReach();
+    }
+
+    /**
+     * @deprecated use {@link #getBlockReach()}
+     */
+    default double getReachDistance()
```

Comment:
These need proper annotations, and I prefer these renames to be right next to the newly named function. It makes it easier for people to find and for me to see that they arnt binary breaking changes.